pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -918,10 +918,12 @@ func ProviderFromSource(
 	result, err := packageresolution.Resolve(
 		pctx.Base(),
 		reg,
+		packageresolution.DefaultWorkspace(),
 		pluginSpec,
 		packageresolution.Options{
-			DisableRegistryResolve: env.DisableRegistryResolve.Value(),
-			Experimental:           env.Experimental.Value(),
+			DisableRegistryResolve:      env.DisableRegistryResolve.Value(),
+			Experimental:                env.Experimental.Value(),
+			IncludeInstalledInWorkspace: true,
```

Comment:
This is in accordance with current behavior for all commands that go through this code path: https://github.com/pulumi/pulumi/issues/20450